src/parse.rs

```diff
@@ -430,48 +430,30 @@ pub(crate) async fn parse_metadata<T: Stream<Item = io::Result<ResponseData>> +
 
 // check if this is simply a unilateral server response
 // (see Section 7 of RFC 3501):
-pub(crate) async fn handle_unilateral(
+pub(crate) fn handle_unilateral(
     res: ResponseData,
     unsolicited: channel::Sender<UnsolicitedResponse>,
 ) {
-    // ignore these if they are not being consumed
-    if unsolicited.is_full() {
-        return;
-    }
-
     match res.parsed() {
         Response::MailboxData(MailboxDatum::Status { mailbox, status }) => {
             unsolicited
-                .send(UnsolicitedResponse::Status {
+                .try_send(UnsolicitedResponse::Status {
                     mailbox: (mailbox.as_ref()).into(),
                     attributes: status.to_vec(),
                 })
-                .await
-                .expect("Channel closed unexpectedly");
+                .ok();
```

Comment:
Would be nice to put a comment along the lines of:

```suggestion
                // If the channel is full or closed,
                // i.e. the responses are not being consumed,
                // ignore new responses.
                .ok();
```

either here or to the top of the function.